plugins/CoreHome/angularjs/comparisons/comparisons.service.js

```diff
@@ -0,0 +1,352 @@
+
+            // add base comparisons
+            compareSegments.unshift(piwikUrl.getSearchParam('segment'));
+            comparePeriods.unshift(piwikUrl.getSearchParam('period'));
+            compareDates.unshift(piwikUrl.getSearchParam('date'));
+
+            var newSegmentComparisons = [];
+            compareSegments.forEach(function (segment, idx) {
+                var storedSegment = null;
+
+                availableSegments.forEach(function (s) {
+                    if (s.definition === segment
+                        || s.definition === decodeURIComponent(segment)
+                        || decodeURIComponent(s.definition) === segment
+                    ) {
+                        storedSegment = s;
+                    }
+                });
+
+                var segmentTitle = storedSegment ? storedSegment.name : _pk_translate('General_Unknown');
+                if (segment.trim() === '') {
+                    segmentTitle = _pk_translate('SegmentEditor_DefaultAllVisits');
+                }
+
+                newSegmentComparisons.push({
+                    params: {
+                        segment: segment
+                    },
+                    title: piwikHelper.htmlDecode(segmentTitle),
+                    index: idx
+                });
+            });
+
+            var newPeriodComparisons = [];
+            for (var i = 0; i < Math.min(compareDates.length, comparePeriods.length); ++i) {
+                try {
+                    title = piwikPeriods.parse(comparePeriods[i], compareDates[i]).getPrettyString();
+                } catch (e) {
+                    title = _pk_translate('General_Error');
+                }
+
+                newPeriodComparisons.push({
+                    params: {
+                        date: compareDates[i],
+                        period: comparePeriods[i]
+                    },
+                    title: title,
+                    index: i
+                });
+            }
+
+            checkEnabledForCurrentPage();
+            setComparisons(newSegmentComparisons, newPeriodComparisons);
+        }
+
+        function setComparisons(newSegmentComparisons, newPeriodComparisons) {
+            var oldSegmentComparisons = segmentComparisons;
+            var oldPeriodComparisons = periodComparisons;
+
+            segmentComparisons = newSegmentComparisons;
+            Object.freeze(segmentComparisons);
+
+            periodComparisons = newPeriodComparisons;
+            Object.freeze(periodComparisons);
+
+            if (JSON.stringify(oldPeriodComparisons) !== JSON.stringify(periodComparisons)
+                || JSON.stringify(oldSegmentComparisons) !== JSON.stringify(segmentComparisons)
+            ) {
+                $rootScope.$emit('piwikComparisonsChanged');
+            }
+        }
+
+        function checkEnabledForCurrentPage() {
+            var category = piwikUrl.getSearchParam('category') || piwikUrl.getSearchParam('module');
```

Comment:
could you leave a comment when the module/action applies? I suppose it applies when it is widgetized?

Reply:
It's for MultiSites, since it uses a module/action. I guess anything that links through the top bar.